packages/app/ui/components/draftInputs/LinkInput.tsx

```diff
@@ -83,6 +88,16 @@ export function LinkInput({ editingPost, isPosting, onSave }: LinkInputProps) {
 
   const form = watch();
   const url = useDebouncedValue(form.url, 500);
+  // We need to track debounce dirty state to appropriately disable the send button. useLinkGrabber's
+  // loading state will not be set until the debounce fires.
+  const [isPendingDebounce, setIsPendingDebounce] = useState(false);
+  useEffect(() => {
+    if (form.url !== url) {
+      setIsPendingDebounce(true);
+    } else {
+      setIsPendingDebounce(false);
+    }
```

Comment:
Seems like it'd be better if `useDebouncedValue` just supported triggering on the lead

Reply:
true

Reply:
```suggestion
  // We need to track debounce dirty state to appropriately disable the send button. useLinkGrabber's
  // loading state will not be set until the debounce fires.
  const [isPendingDebounce, setIsPendingDebounce] = useState(false);
  useEffect(() => {
    setIsPendingDebounce(form.url !== url);
```